### PR TITLE
add mu_different

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: tests/mu_different_test.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+#include "macro_utils/macro_utils.h"
+
+#include "mu_different_test.h"
+
+int run_mu_different_tests(void)
+{
+    int result = 0;
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(-2) != -22);
```
Comment (dcristoloveanu) [inline tests/mu_different_test.c:15]: >-22 [](http://example.com/codeflow?start=41&length=3)

A wild -22 appears. I smell fat fingers.

File: inc/macro_utils/macro_utils.h [hunk 2]
```diff
@@ -357,6 +357,15 @@ __pragma(warning(pop))
...
 
 
+#define MU_IS_VALUE_EQUAL_ONE_OF_VALUES(VALUE, ONE_OF_VALUES) +((VALUE)==(ONE_OF_VALUES))
+
+/*evaluates to 0 or 1*/
+#define MU_IS_NONE_OF(VALUE, ...) ( MU_FOR_EACH_1_KEEP_1(MU_IS_VALUE_EQUAL_ONE_OF_VALUES, (VALUE), __VA_ARGS__) == 0)
+
+#define MU_IS_NONE_OF_EXPRESSION_BUILDER(N, ...) MU_IS_NONE_OF(N, __VA_ARGS__) ? (N) :
+
+#define MU_DIFFERENT(...) (MU_DO(MU_COUNT_ARG(__VA_ARGS__), MU_IS_NONE_OF_EXPRESSION_BUILDER, __VA_ARGS__) - 1) /*-1 is the last ?: ternary third operator, and it is never an output of this macro*/
```
Comment (dcristoloveanu) [inline inc/macro_utils/macro_utils.h:367]: >- 1 [](http://example.com/codeflow?start=107&length=3)

/consider I would write this without the space (-1)

File: inc/macro_utils/macro_utils.h [hunk 3]
```diff
@@ -357,6 +357,15 @@ __pragma(warning(pop))
 #endif
 
 
+#define MU_IS_VALUE_EQUAL_ONE_OF_VALUES(VALUE, ONE_OF_VALUES) +((VALUE)==(ONE_OF_VALUES))
+
+/*evaluates to 0 or 1*/
+#define MU_IS_NONE_OF(VALUE, ...) ( MU_FOR_EACH_1_KEEP_1(MU_IS_VALUE_EQUAL_ONE_OF_VALUES, (VALUE), __VA_ARGS__) == 0)
+
```
Comment (dcristoloveanu) [inline inc/macro_utils/macro_utils.h:364]: A rather enjoyable review

File: tests/mu_different_test.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    /*works with floats too*/
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0.5) != 0.5)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0.5, 0) != 0.5) &&
+        (MU_DIFFERENT(0.5, 0) != 0)
+    );
```
Comment (dcristoloveanu) [inline tests/mu_different_test.c:59]: Consider writing a test with some functions being called. Yes, I know we said side effects are not advised, but for the fun of it.

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: inc/macro_utils/macro_utils.h (hunk 2 above)
Comment (anporumb) [reply inc/macro_utils/macro_utils.h:367]: Extremely fair observation. Just to indicate that we're talking here about number -1 I put it all in paranthesis.

File: tests/mu_different_test.c (hunk 1 above)
Comment (anporumb) [reply tests/mu_different_test.c]: Fat fingers have been retracted.

**Pull request merged**